pkgs/top-level/all-packages.nix

```diff
@@ -19103,7 +19106,7 @@ with pkgs;
   grpc = callPackage ../development/libraries/grpc {
     # grpc builds with c++14 so abseil must also be built that way
     abseil-cpp = abseil-cpp_202111.override {
-      cxxStandard = "14";
+      cxxStandard = "17";
```

Comment:
C++17 is the default standard of current `stdenv.cc` on linux and darwin. Passing it explicitly changes the derivation and installs a copy of abseil that is identical to the default version under a new path. It's probably not an issue, but still a bit weird.

Also, the current [staging next PR](https://github.com/NixOS/nixpkgs/pull/198236) contains a grpc bump that changes the abseil version to `202206`.

Reply:
> C++17 is the default standard of current stdenv.cc on linux and darwin.

Ah, great. I can clean some things up then.

> Also, the current https://github.com/NixOS/nixpkgs/pull/198236 contains a grpc bump that changes the abseil version to 202206.

I guess we should wait to merge this PR until the staging-next PR is merged then so that I can update here as needed.

Reply:
I removed the explicit passing of `cxxStandard` for arrow-cpp, grpc, and google-cloud-cpp